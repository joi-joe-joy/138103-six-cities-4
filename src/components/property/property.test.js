import React from "react";
import renderer from "react-test-renderer";
import Property from "./property.jsx";
import {Provider} from "react-redux";
import configureStore from "redux-mock-store";

const mockStore = configureStore([]);

const offerWithPremium = {
  title: `Canal View Prinsengracht`,
  premium: true,
  pictures: [
    `img/apartment-01.jpg`,
    `img/apartment-02.jpg`,
    `img/studio-photos.jpg`
  ],
  price: 280,
  rating: 3.8,
  type: `house`,
  bedrooms: 3,
  amenities: [`Indoor fireplace`, `Kitchen`, `Wifi`, `Washer`],
  maxGuestsNumber: `Max 2 adults, 1 children`,
  description: `The apartment has a Queen size bed and a very wide modern couch that turns into a bed for a single person by moving the single portion of the couch to the bottom part of the love seat portion of the couch to create a long comfortable bed for a 3rd person. I hope you will enjoy the decor, it has a european feel towards the French side. You will be in the heart of the East Village in Manhattan, considered to be one of the most exciting and dynamic neighborhoods of the world.`,
  host: {
    photo: ``,
    name: `John Donn`,
    super: true
  },
  reviews: [],
  coordinations: [52.3909553943508, 4.929309666406198],
  nearOffers: [
    {
      id: 2,
      title: `Wood and stone place`,
      premium: false,
      pictures: [
        `img/apartment-01.jpg`,
        `img/apartment-02.jpg`,
        `img/apartment-03.jpg`,
        `img/room.jpg`,
        `img/studio-01.jpg`,
        `img/studio-photos.jpg`
      ],
      price: 180,
      rating: 4.3,
      type: `hotel`,
      bedrooms: 2,
      amenities: [`Indoor fireplace`, `Kitchen`, `Wifi`, `Washer`],
      maxGuestsNumber: `Max 2 adults, 2 children`,
      description: `Newly renovated quiet cozy haven in the middle of the hustle and bustle of New York City with breathtaking views of the Manhattan skyline from every window.`,
      host: {
        photo: `img/avatar-max.jpg`,
        name: `Maxim Doff`,
        super: false
      },
      coordinations: [52.369553943508, 4.85309666406198],
      reviews: [
        {
          id: 5,
          author: {
            photo: `img/avatar-angelina.jpg`,
            name: `Monica`
          },
          description: `bedrooms – couples and groups of friends will find this accommodation`,
          date: 1593095836000,
          rating: 3
        },
        {
          id: 6,
          author: {
            photo: `img/avatar-angelina.jpg`,
            name: `Bell Gover`
          },
          description: `friends will find this accommodation`,
          date: 1603095836000,
          rating: 5
        }
      ]
    }
  ]
};

const offerWithoutPremium = {
  title: `Canal View Prinsengracht`,
  premium: true,
  pictures: [
    `img/apartment-01.jpg`,
    `img/apartment-02.jpg`,
    `img/studio-photos.jpg`
  ],
  price: 280,
  rating: 3.8,
  type: `house`,
  bedrooms: 3,
  amenities: [`Indoor fireplace`, `Kitchen`, `Wifi`, `Washer`],
  maxGuestsNumber: `Max 2 adults, 1 children`,
  description: `The apartment has a Queen size bed and a very wide modern couch that turns into a bed for a single person by moving the single portion of the couch to the bottom part of the love seat portion of the couch to create a long comfortable bed for a 3rd person. I hope you will enjoy the decor, it has a european feel towards the French side. You will be in the heart of the East Village in Manhattan, considered to be one of the most exciting and dynamic neighborhoods of the world.`,
  host: {
    photo: ``,
    name: `John Donn`,
    super: true
  },
  reviews: [],
  coordinations: [52.369553943508, 4.85309666406198],
  nearOffers: [
    {
      id: 2,
      title: `Wood and stone place`,
      premium: false,
      pictures: [
        `img/apartment-01.jpg`,
        `img/apartment-02.jpg`,
        `img/apartment-03.jpg`,
        `img/room.jpg`,
        `img/studio-01.jpg`,
        `img/studio-photos.jpg`
      ],
      price: 180,
      rating: 4.3,
      type: `hotel`,
      bedrooms: 2,
      amenities: [`Indoor fireplace`, `Kitchen`, `Wifi`, `Washer`],
      maxGuestsNumber: `Max 2 adults, 2 children`,
      description: `Newly renovated quiet cozy haven in the middle of the hustle and bustle of New York City with breathtaking views of the Manhattan skyline from every window.`,
      host: {
        photo: `img/avatar-max.jpg`,
        name: `Maxim Doff`,
        super: false
      },
      coordinations: [52.369553943508, 4.85309666406198],
      reviews: [
        {
          id: 5,
          author: {
            photo: `img/avatar-angelina.jpg`,
            name: `Monica`
          },
          description: `bedrooms – couples and groups of friends will find this accommodation`,
          date: 1593095836000,
          rating: 3
        },
        {
          id: 6,
          author: {
            photo: `img/avatar-angelina.jpg`,
            name: `Bell Gover`
          },
          description: `friends will find this accommodation`,
          date: 1603095836000,
          rating: 5
        }
      ]
    }
  ]
};

it(`Render Property correctly`, () => {
  const store = mockStore({
    card: {}
  });

  const tree = renderer.create(
      <Provider store={store}>
        <Property
          offer={offerWithPremium}/>
      </Provider>
  ).toJSON();

  expect(tree).toMatchSnapshot();
});

it(`Property render correctly without Premium`, () => {
  const store = mockStore({
    card: {}
  });

  const tree = renderer
    .create(
        <Provider store={store}>
          <Property
            offer={offerWithoutPremium}/>
        </Provider>
    )
    .toJSON();

  expect(tree).toMatchSnapshot();
});
