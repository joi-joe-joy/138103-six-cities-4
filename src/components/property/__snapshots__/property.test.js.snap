// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Property render correctly without Premium 1`] = `
<main
  className="page__main page__main--property"
>
  <section
    className="property"
  >
    <div
      className="property__gallery-container container"
    >
      <div
        className="property__gallery"
      >
        <div
          className="property__image-wrapper"
        >
          <img
            alt="Canal View Prinsengracht"
            className="property__image"
            src="img/apartment-01.jpg"
          />
        </div>
        <div
          className="property__image-wrapper"
        >
          <img
            alt="Canal View Prinsengracht"
            className="property__image"
            src="img/apartment-02.jpg"
          />
        </div>
        <div
          className="property__image-wrapper"
        >
          <img
            alt="Canal View Prinsengracht"
            className="property__image"
            src="img/studio-photos.jpg"
          />
        </div>
      </div>
    </div>
    <div
      className="property__container container"
    >
      <div
        className="property__wrapper"
      >
        <div
          className="property__mark"
        >
          <span>
            Premium
          </span>
        </div>
        <div
          className="property__name-wrapper"
        >
          <h1
            className="property__name"
          >
            Canal View Prinsengracht
          </h1>
          <button
            className="property__bookmark-button button"
            type="button"
          >
            <svg
              height="33"
              width="31"
            />
            <span
              className="visually-hidden"
            >
              To bookmarks
            </span>
          </button>
        </div>
        <div
          className="property__rating rating"
        >
          <div
            className="property__stars rating__stars"
          >
            <span
              style={
                Object {
                  "width": "76%",
                }
              }
            />
            <span
              className="visually-hidden"
            >
              Rating
            </span>
          </div>
          <span
            className="property__rating-value rating__value"
          >
            3.8
          </span>
        </div>
        <ul
          className="property__features"
        >
          <li
            className="property__feature property__feature--entire"
          >
            House
          </li>
          <li
            className="property__feature property__feature--bedrooms"
          >
            3
             
            Bedroom
          </li>
          <li
            className="property__feature property__feature--adults"
          >
            Max 2 adults, 1 children
          </li>
        </ul>
        <div
          className="property__price"
        >
          <b
            className="property__price-value"
          >
            €
            280
          </b>
          <span
            className="property__price-text"
          >
             night
          </span>
        </div>
        <div
          className="property__inside"
        >
          <h2
            className="property__inside-title"
          >
            What's inside
          </h2>
          <ul
            className="property__inside-list"
          >
            <li
              className="property__inside-item"
            >
              Indoor fireplace
            </li>
            <li
              className="property__inside-item"
            >
              Kitchen
            </li>
            <li
              className="property__inside-item"
            >
              Wifi
            </li>
            <li
              className="property__inside-item"
            >
              Washer
            </li>
          </ul>
        </div>
        <div
          className="property__host"
        >
          <h2
            className="property__host-title"
          >
            Meet the host
          </h2>
          <div
            className="property__host-user user"
          >
            <div
              className="property__avatar-wrapper user__avatar-wrapper property__avatar-wrapper--pro"
            >
              <img
                alt="Host avatar"
                className="property__avatar user__avatar"
                height="74"
                src="img/avatar.svg"
                width="74"
              />
            </div>
            <span
              className="property__user-name"
            >
              John Donn
            </span>
          </div>
          <div
            className="property__description"
          >
            <p
              className="property__text"
            >
              The apartment has a Queen size bed and a very wide modern couch that turns into a bed for a single person by moving the single portion of the couch to the bottom part of the love seat portion of the couch to create a long comfortable bed for a 3rd person. I hope you will enjoy the decor, it has a european feel towards the French side. You will be in the heart of the East Village in Manhattan, considered to be one of the most exciting and dynamic neighborhoods of the world.
            </p>
          </div>
        </div>
        <section
          className="property__reviews reviews"
        >
          <h2
            className="reviews__title"
          >
            Reviews ·
            <span
              className="reviews__amount"
            >
              0
            </span>
          </h2>
          <ul
            className="reviews__list"
          />
          <form
            action="#"
            className="reviews__form form"
            method="post"
          >
            <label
              className="reviews__label form__label"
              htmlFor="review"
            >
              Your review
            </label>
            <div
              className="reviews__rating-form form__rating"
            >
              <input
                className="form__rating-input visually-hidden"
                id="5-stars"
                name="rating"
                type="radio"
                value="5"
              />
              <label
                className="reviews__rating-label form__rating-label"
                htmlFor="5-stars"
                title="perfect"
              >
                <svg
                  height="33"
                  width="37"
                />
              </label>
              <input
                className="form__rating-input visually-hidden"
                id="4-stars"
                name="rating"
                type="radio"
                value="4"
              />
              <label
                className="reviews__rating-label form__rating-label"
                htmlFor="4-stars"
                title="good"
              >
                <svg
                  height="33"
                  width="37"
                />
              </label>
              <input
                className="form__rating-input visually-hidden"
                id="3-stars"
                name="rating"
                type="radio"
                value="3"
              />
              <label
                className="reviews__rating-label form__rating-label"
                htmlFor="3-stars"
                title="not bad"
              >
                <svg
                  height="33"
                  width="37"
                />
              </label>
              <input
                className="form__rating-input visually-hidden"
                id="2-stars"
                name="rating"
                type="radio"
                value="2"
              />
              <label
                className="reviews__rating-label form__rating-label"
                htmlFor="2-stars"
                title="badly"
              >
                <svg
                  height="33"
                  width="37"
                />
              </label>
              <input
                className="form__rating-input visually-hidden"
                id="1-star"
                name="rating"
                type="radio"
                value="1"
              />
              <label
                className="reviews__rating-label form__rating-label"
                htmlFor="1-star"
                title="terribly"
              >
                <svg
                  height="33"
                  width="37"
                />
              </label>
            </div>
            <textarea
              className="reviews__textarea form__textarea"
              id="review"
              name="review"
              placeholder="Tell how was your stay, what you like and what can be improved"
            />
            <div
              className="reviews__button-wrapper"
            >
              <p
                className="reviews__help"
              >
                To submit review please make sure to set 
                <span
                  className="reviews__star"
                >
                  rating
                </span>
                 and describe your stay with at least 
                <b
                  className="reviews__text-amount"
                >
                  50 characters
                </b>
                .
              </p>
              <button
                className="reviews__submit form__submit button"
                disabled=""
                type="submit"
              >
                Submit
              </button>
            </div>
          </form>
        </section>
      </div>
    </div>
    <section
      className="map property__map"
    >
      <div
        id="map"
        style={
          Object {
            "height": "100%",
          }
        }
      />
    </section>
  </section>
  <div
    className="container"
  >
    <section
      className="near-places places"
    >
      <h2
        className="near-places__title"
      >
        Other places in the neighbourhood
      </h2>
      <div
        className="places__list near-places__list"
      >
        <article
          className="place-card near-places__card"
          onMouseOver={[Function]}
        >
          <div
            className="place-card__image-wrapper near-places__image-wrapper"
          >
            <a
              href="#"
            >
              <img
                alt="Wood and stone place"
                className="place-card__image"
                height="200"
                src="img/apartment-01.jpg"
                width="260"
              />
            </a>
          </div>
          <div
            className="place-card__info"
          >
            <div
              className="place-card__price-wrapper"
            >
              <div
                className="place-card__price"
              >
                <b
                  className="place-card__price-value"
                >
                  €
                  180
                </b>
                <span
                  className="place-card__price-text"
                >
                  / night
                </span>
              </div>
              <button
                className="place-card__bookmark-button button"
                onClick={[Function]}
                type="button"
              >
                <svg
                  height="18"
                  width="17"
                />
                <span
                  className="visually-hidden"
                >
                  To bookmarks
                </span>
              </button>
            </div>
            <div
              className="place-card__rating rating"
            >
              <div
                className="place-card__stars rating__stars"
              >
                <span
                  style={
                    Object {
                      "width": "80%",
                    }
                  }
                />
                <span
                  className="visually-hidden"
                >
                  Rating
                </span>
              </div>
            </div>
            <h2
              className="place-card__name"
            >
              <a
                onClick={[Function]}
              >
                Wood and stone place
              </a>
            </h2>
            <p
              className="place-card__type"
            >
              Hotel
            </p>
          </div>
        </article>
      </div>
    </section>
  </div>
</main>
`;

exports[`Render Property correctly 1`] = `
<main
  className="page__main page__main--property"
>
  <section
    className="property"
  >
    <div
      className="property__gallery-container container"
    >
      <div
        className="property__gallery"
      >
        <div
          className="property__image-wrapper"
        >
          <img
            alt="Canal View Prinsengracht"
            className="property__image"
            src="img/apartment-01.jpg"
          />
        </div>
        <div
          className="property__image-wrapper"
        >
          <img
            alt="Canal View Prinsengracht"
            className="property__image"
            src="img/apartment-02.jpg"
          />
        </div>
        <div
          className="property__image-wrapper"
        >
          <img
            alt="Canal View Prinsengracht"
            className="property__image"
            src="img/studio-photos.jpg"
          />
        </div>
      </div>
    </div>
    <div
      className="property__container container"
    >
      <div
        className="property__wrapper"
      >
        <div
          className="property__mark"
        >
          <span>
            Premium
          </span>
        </div>
        <div
          className="property__name-wrapper"
        >
          <h1
            className="property__name"
          >
            Canal View Prinsengracht
          </h1>
          <button
            className="property__bookmark-button button"
            type="button"
          >
            <svg
              height="33"
              width="31"
            />
            <span
              className="visually-hidden"
            >
              To bookmarks
            </span>
          </button>
        </div>
        <div
          className="property__rating rating"
        >
          <div
            className="property__stars rating__stars"
          >
            <span
              style={
                Object {
                  "width": "76%",
                }
              }
            />
            <span
              className="visually-hidden"
            >
              Rating
            </span>
          </div>
          <span
            className="property__rating-value rating__value"
          >
            3.8
          </span>
        </div>
        <ul
          className="property__features"
        >
          <li
            className="property__feature property__feature--entire"
          >
            House
          </li>
          <li
            className="property__feature property__feature--bedrooms"
          >
            3
             
            Bedroom
          </li>
          <li
            className="property__feature property__feature--adults"
          >
            Max 2 adults, 1 children
          </li>
        </ul>
        <div
          className="property__price"
        >
          <b
            className="property__price-value"
          >
            €
            280
          </b>
          <span
            className="property__price-text"
          >
             night
          </span>
        </div>
        <div
          className="property__inside"
        >
          <h2
            className="property__inside-title"
          >
            What's inside
          </h2>
          <ul
            className="property__inside-list"
          >
            <li
              className="property__inside-item"
            >
              Indoor fireplace
            </li>
            <li
              className="property__inside-item"
            >
              Kitchen
            </li>
            <li
              className="property__inside-item"
            >
              Wifi
            </li>
            <li
              className="property__inside-item"
            >
              Washer
            </li>
          </ul>
        </div>
        <div
          className="property__host"
        >
          <h2
            className="property__host-title"
          >
            Meet the host
          </h2>
          <div
            className="property__host-user user"
          >
            <div
              className="property__avatar-wrapper user__avatar-wrapper property__avatar-wrapper--pro"
            >
              <img
                alt="Host avatar"
                className="property__avatar user__avatar"
                height="74"
                src="img/avatar.svg"
                width="74"
              />
            </div>
            <span
              className="property__user-name"
            >
              John Donn
            </span>
          </div>
          <div
            className="property__description"
          >
            <p
              className="property__text"
            >
              The apartment has a Queen size bed and a very wide modern couch that turns into a bed for a single person by moving the single portion of the couch to the bottom part of the love seat portion of the couch to create a long comfortable bed for a 3rd person. I hope you will enjoy the decor, it has a european feel towards the French side. You will be in the heart of the East Village in Manhattan, considered to be one of the most exciting and dynamic neighborhoods of the world.
            </p>
          </div>
        </div>
        <section
          className="property__reviews reviews"
        >
          <h2
            className="reviews__title"
          >
            Reviews ·
            <span
              className="reviews__amount"
            >
              0
            </span>
          </h2>
          <ul
            className="reviews__list"
          />
          <form
            action="#"
            className="reviews__form form"
            method="post"
          >
            <label
              className="reviews__label form__label"
              htmlFor="review"
            >
              Your review
            </label>
            <div
              className="reviews__rating-form form__rating"
            >
              <input
                className="form__rating-input visually-hidden"
                id="5-stars"
                name="rating"
                type="radio"
                value="5"
              />
              <label
                className="reviews__rating-label form__rating-label"
                htmlFor="5-stars"
                title="perfect"
              >
                <svg
                  height="33"
                  width="37"
                />
              </label>
              <input
                className="form__rating-input visually-hidden"
                id="4-stars"
                name="rating"
                type="radio"
                value="4"
              />
              <label
                className="reviews__rating-label form__rating-label"
                htmlFor="4-stars"
                title="good"
              >
                <svg
                  height="33"
                  width="37"
                />
              </label>
              <input
                className="form__rating-input visually-hidden"
                id="3-stars"
                name="rating"
                type="radio"
                value="3"
              />
              <label
                className="reviews__rating-label form__rating-label"
                htmlFor="3-stars"
                title="not bad"
              >
                <svg
                  height="33"
                  width="37"
                />
              </label>
              <input
                className="form__rating-input visually-hidden"
                id="2-stars"
                name="rating"
                type="radio"
                value="2"
              />
              <label
                className="reviews__rating-label form__rating-label"
                htmlFor="2-stars"
                title="badly"
              >
                <svg
                  height="33"
                  width="37"
                />
              </label>
              <input
                className="form__rating-input visually-hidden"
                id="1-star"
                name="rating"
                type="radio"
                value="1"
              />
              <label
                className="reviews__rating-label form__rating-label"
                htmlFor="1-star"
                title="terribly"
              >
                <svg
                  height="33"
                  width="37"
                />
              </label>
            </div>
            <textarea
              className="reviews__textarea form__textarea"
              id="review"
              name="review"
              placeholder="Tell how was your stay, what you like and what can be improved"
            />
            <div
              className="reviews__button-wrapper"
            >
              <p
                className="reviews__help"
              >
                To submit review please make sure to set 
                <span
                  className="reviews__star"
                >
                  rating
                </span>
                 and describe your stay with at least 
                <b
                  className="reviews__text-amount"
                >
                  50 characters
                </b>
                .
              </p>
              <button
                className="reviews__submit form__submit button"
                disabled=""
                type="submit"
              >
                Submit
              </button>
            </div>
          </form>
        </section>
      </div>
    </div>
    <section
      className="map property__map"
    >
      <div
        id="map"
        style={
          Object {
            "height": "100%",
          }
        }
      />
    </section>
  </section>
  <div
    className="container"
  >
    <section
      className="near-places places"
    >
      <h2
        className="near-places__title"
      >
        Other places in the neighbourhood
      </h2>
      <div
        className="places__list near-places__list"
      >
        <article
          className="place-card near-places__card"
          onMouseOver={[Function]}
        >
          <div
            className="place-card__image-wrapper near-places__image-wrapper"
          >
            <a
              href="#"
            >
              <img
                alt="Wood and stone place"
                className="place-card__image"
                height="200"
                src="img/apartment-01.jpg"
                width="260"
              />
            </a>
          </div>
          <div
            className="place-card__info"
          >
            <div
              className="place-card__price-wrapper"
            >
              <div
                className="place-card__price"
              >
                <b
                  className="place-card__price-value"
                >
                  €
                  180
                </b>
                <span
                  className="place-card__price-text"
                >
                  / night
                </span>
              </div>
              <button
                className="place-card__bookmark-button button"
                onClick={[Function]}
                type="button"
              >
                <svg
                  height="18"
                  width="17"
                />
                <span
                  className="visually-hidden"
                >
                  To bookmarks
                </span>
              </button>
            </div>
            <div
              className="place-card__rating rating"
            >
              <div
                className="place-card__stars rating__stars"
              >
                <span
                  style={
                    Object {
                      "width": "80%",
                    }
                  }
                />
                <span
                  className="visually-hidden"
                >
                  Rating
                </span>
              </div>
            </div>
            <h2
              className="place-card__name"
            >
              <a
                onClick={[Function]}
              >
                Wood and stone place
              </a>
            </h2>
            <p
              className="place-card__type"
            >
              Hotel
            </p>
          </div>
        </article>
      </div>
    </section>
  </div>
</main>
`;
